### PR TITLE
Fix prompted values section of schedule details

### PR DESCRIPTION
**Pull request by mabashian**
##### SUMMARY
link #6510 

Talked with @marshmalien and we came up with this UX pattern:

Screen Shot 2020-12-11 at 11 11 48 AM

We think that this most closely matches the direction that PF is going with subforms.

I made the same UX changes to the PromptDetail component since we also show prompted values in a similar way in that component.

@one-t @unlikelyzero @tiagodread - to whoever tests this.  Note that prompting on the schedules form is not currently supported.  See #7692.  So you'll have to interact with the api to set this up.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
 - UI

### CODE REVIEW

File: awx/ui_next/src/components/Schedule/ScheduleDetail/ScheduleDetail.jsx [hunk 1]
```diff
@@ -140,18 +151,25 @@ function ScheduleDetail({ schedule, i18n }) {
     survey_enabled,
   } = launchData || {};
 
+  const showCredentialsDetail =
+    ask_credential_on_launch && credentials.length > 0;
+  const showInventoryDetail = ask_inventory_on_launch && inventory;
```
Comment (marshmalien) [inline awx/ui_next/src/components/Schedule/ScheduleDetail/ScheduleDetail.jsx:165]: I like these additions. Do you think we should do this for other fields like `ask_skip_tags_on_launch` and `ask_tags_on_launch`?

I created a job template with zero skip tags and selected skip tags prompt on launch. Below we show a Prompted Values section, but no skip tags detail field. Do we want to show an empty Prompted Values section?
Screen Shot 2020-12-11 at 4 17 54 PM

File: awx/ui_next/src/components/PromptDetail/PromptDetail.jsx [hunk 2]
```diff
@@ -18,9 +18,19 @@ import PromptInventorySourceDetail from './PromptInventorySourceDetail';
 import PromptJobTemplateDetail from './PromptJobTemplateDetail';
 import PromptWFJobTemplateDetail from './PromptWFJobTemplateDetail';
 
-const PromptHeader = styled.h2`
-  font-weight: bold;
-  margin: var(--pf-global--spacer--lg) 0;
+const PromptTitle = styled(Title)`
+  margin-top: 40px;
```
Comment (marshmalien) [inline awx/ui_next/src/components/PromptDetail/PromptDetail.jsx]:
```suggestion
  margin-top: var(--pf-global--spacer--xl);
```
We could make them all PatternFly utility variables. Either way works for me.

File: awx/ui_next/src/components/PromptDetail/PromptDetail.jsx [hunk 3]
```diff
@@ -18,9 +18,19 @@ import PromptInventorySourceDetail from './PromptInventorySourceDetail';
...
+  grid-column: 1 / -1;
+`;
+
+const PromptDivider = styled(Divider)`
+  margin-top: var(--pf-global--spacer--lg);
+  margin-bottom: var(--pf-global--spacer--lg);
+`;
+
+const PromptDetailList = styled(DetailList)`
+  padding: 0px 20px;
```
Comment (marshmalien) [inline awx/ui_next/src/components/PromptDetail/PromptDetail.jsx]:
```suggestion
  padding: 0px var(--pf-global--spacer--lg);
```

File: awx/ui_next/src/components/Schedule/ScheduleDetail/ScheduleDetail.jsx (hunk 1 above)
Comment (mabashian) [reply awx/ui_next/src/components/Schedule/ScheduleDetail/ScheduleDetail.jsx:165]: Oof, yes I'll make those same changes for those fields.  Good catch.

**Comment by mabashian** [comment]
OK @marshmalien I think this is ready for another look

**Comment by mabashian** [comment]
Before we merge this I need to write better tests around checking for when the `Prompted Values` section is hidden/shown.

**Review by marshmalien — APPROVED** [review]
🧨 Tested out the latest changes locally and this look good to me. Thanks for adding test coverage for those two additional cases!

**Comment by one-t** [comment]
This looks good to me- we have a stub for this already in tower-qe.
image

**Pull request merged**